pkg/scheduler/core/spreadconstraint/group_clusters.go

```diff
@@ -136,26 +136,29 @@ func (info *GroupClustersInfo) generateZoneInfo(spreadConstraints []policyv1alph
 	}
 
 	for _, clusterInfo := range info.Clusters {
-		zone := clusterInfo.Cluster.Spec.Zone
-		if zone == "" {
+		zones := clusterInfo.Cluster.Spec.Zones
+		if len(zones) == 0 {
 			continue
 		}
 
-		zoneInfo, ok := info.Zones[zone]
-		if !ok {
-			zoneInfo = ZoneInfo{
-				Name:     zone,
-				Clusters: make([]ClusterDetailInfo, 0),
+		for _, zone := range zones {
+			zoneInfo, ok := info.Zones[zone]
+			if !ok {
+				zoneInfo = ZoneInfo{
+					Name:     zone,
+					Clusters: make([]ClusterDetailInfo, 0),
+				}
 			}
+			zoneInfo.Clusters = append(zoneInfo.Clusters, clusterInfo)
+			zoneInfo.AvailableReplicas += clusterInfo.AvailableReplicas
+			info.Zones[zone] = zoneInfo
 		}
-
-		zoneInfo.Clusters = append(zoneInfo.Clusters, clusterInfo)
-		zoneInfo.AvailableReplicas += clusterInfo.AvailableReplicas
-		info.Zones[zone] = zoneInfo
 	}
 
 	for zone, zoneInfo := range info.Zones {
-		zoneInfo.Score = zoneInfo.Clusters[0].Score
+		for _, cluster := range zoneInfo.Clusters {
+			zoneInfo.Score += cluster.Score
```

Comment:
I doubt the score should be summed here. 
ZoneInfo.Score indicates the highest score in all clusters of the zone. 

I think it should still be `zoneInfo.Score = zoneInfo.Clusters[0].Score`, I assume the zoneInfo.Clusters already sorted by score.

Reply:
ok